tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_scope_test.dart

```diff
@@ -590,6 +590,38 @@ void main() {
     },
   );
 
+  test(
+    'Given a server only class with a none nullable field with the scope serverOnly then field has scope server only.',
+    () {
+      var models = [
+        ModelSourceBuilder().withYaml(
+          '''
+          class: Example
+          serverOnly: true
+          fields:
+            name: String, scope=serverOnly
```

Comment:
Suggestion: none blocking, this could generate an info "unnecessary" message, this will in VS Code render the text with lower opacity (meaning the user can remove this without changing the behaviour).